src/views/EditUserView.vue

```diff
@@ -0,0 +1,240 @@
+        >
+        <div class="invalid-feedback">
+          Please fill out this field.
+        </div>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <br>
+      <button
+        type="success"
+        name="submit"
+        class="btn btn-primary"
+        style="margin: auto"
+        @click="updateUser"
+      >
+        Register
+      </button>
+    </div>
+    <div
+      v-if="regError"
+      class="alert alert-danger"
+      role="alert"
+      name="regError"
+    >
+      {{ regError }}
+    </div>
+  </form>
+</template>
+
+<script setup>
+  import{ ref, watch } from 'vue';
+  import { useRouter } from 'vue-router';
+  import { firebaseStore, collection, query, where, getDocs, updateDoc, doc } from '../firebase/database';
+  import { getAuth } from '../firebase/firebaseAuth';
+  import { useUserStore } from '../stores/user';
+  const firstName = ref('');
+  const surname = ref('');
+  const address1 = ref('');
+  const address2 = ref('');
+  const address3 = ref('');
+  const city = ref('');
+  const postCode = ref('');
+  const phoneNumber = ref('');
+  const affiliation = ref('');
+  const router = useRouter();
+  const regError = ref('');
+  const auth = getAuth();
+
+
+  const userStore = useUserStore();
+  const user = userStore.user;
+  firstName.value = user.info.firstName;
+  surname.value = user.info.surname;
+  address1.value = user.info.addressline1;
+  address2.value = user.info.addressline2;
+  address3.value = user.info.addressline3;
+  city.value = user.info.city;
+  postCode.value = user.info.postCode;
+  phoneNumber.value = user.info.phoneNumber;
+  affiliation.value = user.info.affiliation;
+
+  watch(phoneNumber, () =>{
+    if(/(^\+?[0-9]{10,15})$/.test(phoneNumber.value) ){
+      regError.value = null;
+    }else{
+      regError.value = 'Phone number is invalid';
+    }
+  });
+
+  async function updateUser(){
+    if(!regError.value){
+      if(!firstName.value || !surname.value || !address1.value|| !city.value || !postCode.value ||!phoneNumber.value
+        || !affiliation.value){
+        regError.value='Please fill all fields';
+      }else{
+        const getUserId = query(collection(firebaseStore,'users_dev'), where('uid','==',auth.currentUser.uid));
+        const getData = await getDocs(getUserId);
+        console.log(getData);
+        getData.forEach(async (docData)=>{
+          try{
+            await updateDoc(doc(firebaseStore, 'user', docData.id),{
+              firstName: firstName.value,
+              surname: surname.value,
+              addressline1: address1.value,
+              addressline2: address2.value,
+              addressline3: address3.value,
+              city: city.value,
+              postCode: postCode.value,
+              phoneNumber: phoneNumber.value,
+              affiliation: affiliation.value
+            });
+            router.push('/profile');
+          }
+          catch(error){
+            console.log(docData.id);
```

Comment:
Either remove document the console logging of the document id to the user since that is sensitive info or, make an if statement that checks if you are VITE_DEV env is true and only console log then. This is something we do not need to expose to the user.

Reply:
Removed catch and console log from this 
![image](https://user-images.githubusercontent.com/46050460/233806006-6fc54d69-2a75-42c7-8848-67f27b37601c.png)

Reply:
console log from? Not sure what the screen shot is supposed to be showing me?

Reply:
Ive removed the console.log of the document data id from the updateUser function
here is how it looks now
![image](https://user-images.githubusercontent.com/46050460/233809088-6a99db1d-3496-42b4-a0bf-2e08563e9f40.png)
It now only logs the error but not the document ID

Reply:
fine